tests/test_arrow_dataset.py

```diff
@@ -1307,12 +1307,19 @@ def test_map_caching(self, in_memory):
             self._caplog.clear()
             with self._caplog.at_level(WARNING):
                 with self._create_dummy_dataset(in_memory, tmp_dir) as dset:
-                    with dset.map(lambda x: {"foo": "bar"}) as dset_test1:
-                        dset_test1_data_files = list(dset_test1.cache_files)
-                    with dset.map(lambda x: {"foo": "bar"}) as dset_test2:
-                        self.assertEqual(dset_test1_data_files, dset_test2.cache_files)
-                        self.assertEqual(len(dset_test2.cache_files), 1 - int(in_memory))
-                        self.assertTrue(("Loading cached processed dataset" in self._caplog.text) ^ in_memory)
+                    with patch(
+                        "datasets.arrow_dataset.Dataset._map_single",
+                        autospec=Dataset._map_single,
```

Comment:
I only added a check for the single-proc mode since `autospec` (preserves `__qualname__` needed for the fingerprint) and `new_callable` (makes the mock object pickable in the multi-proc mode) cannot be specified at the same time.

I think this is fine as we already have a test to check the number of `multiprocess.Pool` calls in `test_map_caching`.

Reply:
Indeed, thanks !